src/main/java/mat/cql/CqlToMatXml.java

```diff
@@ -0,0 +1,609 @@
+            // include NCQA_Common version '5.1.000' called Common
+            String[] words = s.split(" ");
+            if (words.length != 6) {
+                throw new IllegalArgumentException("Invalid include encountered: " + s);
+            }
+            CQLIncludeLibrary result = new CQLIncludeLibrary();
+            result.setId(newGuid());
+            result.setCqlLibraryName(words[1]);
+            result.setVersion(chomp1(words[3]));
+            result.setAliasName(words[5]);
+            result.setQdmVersion(destinationModel.getUsingModelVersion());
+            result.setLibraryModelType("FHIR");
+
+            result.setCqlLibraryId(nameToGlobalLibId.get(result.getCqlLibraryName()));
+            if (result.getCqlLibraryId() == null) {
+                //Non global library case.
+                //We are not handling this in this release so return null for now.
+                //Eventually this will have to convert the library also.
+                result = null;
+            }
+            return result;
+        }));
+    }
+
+    /**
+     * Parses all of the code systems found in the cql.
+     */
+    private void processCodeSystems() {
+        destinationModel.setCodeSystemList(parseRepeatingLine(CODESYSTEM_TOKEN, s -> {
+            //codesystem "LOINC": 'urn:oid:2.16.840.1.113883.6.1'
+            ParseResult systemName = nextQuotedString(s, 0);
+            ParseResult version = nextTickedString(s, systemName.getEndIndex());
+
+            if (areValidAscendingIndexes(systemName, version)) {
+                //For now these should be identical to the sourceCqlModel versions.
+                //Just find the corresponding one by code system name and use that.
+                String sysName = systemName.getString();
+                String ver = trimUrn(version.getString());
+                log.info("sysName=" + sysName + " version=" + ver);
+                return findExisting(sourceModel.getCodeSystemList(),
+                        cs -> StringUtils.equals(systemName.getString(), cs.getCodeSystemName()),
+                        "Could not find sourceCqlModel.codeSystemList for " + sysName);
+            } else {
+                throw new IllegalArgumentException("Invalid include encountered: " + s);
+            }
+        }));
+    }
+
+    /**
+     * Parses all of the valuesets found in the cql.
+     */
+    private void processValueSets() {
+        destinationModel.setValueSetList(parseRepeatingLine(VALUESET_TOKEN, s -> {
+            //valueset "Ethnicity": 'urn:oid:2.16.840.1.114222.4.11.837'
+            ParseResult type = nextQuotedString(s, 0);
+            ParseResult version = nextTickedString(s, type.getEndIndex());
+
+            if (areValidAscendingIndexes(type, version)) {
+                //For now these should be identical to the sourceCqlModel versions.
+                //Just find the corresponding one by code system name and use that.
```

Comment:
Comparing code list name and not code system name.

Reply:
Fixed